routes/insights.js

```diff
@@ -27,33 +28,41 @@ router.get('/', (req, res) => {
   );
 });
 
-router.get('/annotate', async (req, res) => {
-  const ids = await getInsightsAnnotated();
-  res.status(200).json(ids);
-})
+router.get('/annotate', async (req, res, next) => {
+  try {
+    const ids = await getInsightsAnnotated();
+    res.status(200).json(ids);
+  } catch (err) {
+    next(err);
+  }
+});
 
 router.post('/annotate', async (req, res, next) => {
   const {
     body: { insight_id: insightId, annotation },
   } = req;
 
-  res.send('OK');
-
-  const row = await getInsightId(insightId);
-  if (!row) await createInsightId(insightId, annotation);
-  else if (!row.is_annotated) {
-    const columnName = +annotation === 1 ? 'nb_true' : 'nb_false';
-    row[columnName] += 1;
-
-    const confirm = checkConfirmationInsight(row);
-    await updateInsightId(insightId, columnName, row[columnName], confirm);
-    if (confirm) {
-      console.log('CONFIRMATION');
-      // postAnnotate(insightId, annotation).catch((err) => {
-      // TODO: prevoir table temporaire si err
-      //   console.log('postAnnotate: err :>> ', err);
-      // });
+  try {
+    const row = await getInsightId(insightId);
+    if (!row) await createInsightId(insightId, annotation);
+    else if (!row.is_annotated) {
+      const columnName = +annotation === 1 ? 'nb_true' : 'nb_false';
+      row[columnName] += 1;
+
```

Comment:
Je fais encore une remarque de style (écrite après celles ci-dessous), sur l'utilisation de if / else if. L'alternative s'appelle le [return early](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement).

Par exemple on pourrait écrire :
```javascript
try {
  const row = await getInsightId(insightId);
  if (!row) {
    await createInsightId(insightId, annotation);
    return res.sendStatus(204);
  }
  if (row.is_annotated) return res.sendStatus(204);
  // ici plus besoin de else...
  const columnName = +annotation === 1 ? 'nb_true' : 'nb_false';
  row[columnName] += 1;
  // etc...
  
  const confirm = checkConfirmationInsight(row);
  await updateInsightId(insightId, columnName, row[columnName], confirm);
  if (!confirm) return res.sendStatus(204);
  // à nouveau plus besoin d'indenter...
  postAnnotate(insightId, annotation).catch((err) => {
     // ...
  });
```

À nouveau, ici ça reste très lisible, mais le "return early" devient particulièrement intéressant quand le bloc derrière ton `else` ou `else if` comporte beaucoup de lignes... (parfois on a du mal à trouver où se termine un bloc ouvert loin "au dessus").